colorcoder/ColorService.java

```diff
@@ -0,0 +1,26 @@
+package colorcoder;
+
+public class ColorService {
+	
+	public ColorPair getColorFromPairNumber(int pairNumber) {
+		int zeroBasedPairNumber = pairNumber - 1;
+		MajorColor majorColor = (MajorColor)fromIndex(MajorColor.values(),zeroBasedPairNumber /Util.numberOfMinorColors);

```

Comment:
Good abstraction with the IColor interface 👍 
Is there any way you can avoid a cast over here? Because it would be easy to make a typo and cast to MinorColor instead of MajorColor.